crates/core/plugin_sm/src/plugin.rs

```diff
@@ -436,10 +438,29 @@ impl Plugin for ExternalPluginCommand {
         let command = self.command(LIST, None)?;
         let output = self.execute(command, logger).await?;
         if output.status.success() {
-            Ok(deserialize_module_info(
-                self.name.clone(),
-                output.stdout.as_slice(),
-            )?)
+            let config = get_tedge_config().unwrap();
+            let limit = config.query(SoftwarePluginMaxPackagesSetting).unwrap().0;
```

Comment:
To use the new tedge_config is super easy:


```suggestion
            let config = get_new_tedge_config().unwrap();
            let limit = config.software.plugin.max_packages;
```

With just one caveat: the function `get_new_tedge_config()` is not existing yet ;-). It has to be added as a temporary dup of `get_tedge_config()`:

```
/// loads the new tedge config from system default
pub fn get_new_tedge_config() -> Result<new::TEdgeConfig, TEdgeConfigError> {
    let tedge_config_location = TEdgeConfigLocation::default();
    TEdgeConfigRepository::new(tedge_config_location).load_new()
}
```